packages/app/src/app/overmind/namespaces/live/internalActions.ts

```diff
@@ -103,30 +97,35 @@ export const initializeModuleState: Action<any> = (
     effects.live.createClient(moduleShortid, moduleInfo.revision);
 
     // Module has not been saved, so is different
-    const index = state.editor.currentSandbox.modules.findIndex(
+    const module = state.editor.currentSandbox.modules.find(
       m => m.shortid === moduleShortid
     );
 
-    if (index > -1) {
+    if (module) {
       if (moduleInfo.code === null) {
         return;
       }
 
       // If we are getting code that has not been saved, we want to update the actual saved code
       // so that we properly detect "changed files" in the explorer. But we only do this if the
       // savedCode is not set, as we might get multiple initializeModuleState
-      if (
-        !moduleInfo.synced &&
-        state.editor.currentSandbox.modules[index].savedCode == null
-      ) {
-        state.editor.currentSandbox.modules[index].savedCode =
-          state.editor.currentSandbox.modules[index].code;
+      if (!moduleInfo.synced && module.savedCode !== moduleInfo) {
```

Comment:
Isn't `module.savedCode !== moduleInfo` always false?

Reply:
```suggestion
      if (!moduleInfo.synced && !module.savedCode) {
```

Oh, it should be this. Not sure what happened there.

Reply:
```suggestion
      if (!moduleInfo.synced && module.savedCode === null) {
```